src/storage/mod.rs

```diff
@@ -147,11 +148,11 @@ pub trait Transaction: Sync + Send + 'static {
 }
 
 /// An iterator over table in a transaction.
-#[async_trait]
 pub trait TxnIterator: Send {
+    type NextFuture<'a>: Future<Output = StorageResult<Option<DataChunk>>> + Send
```

Comment:
Better to add a `'a` bound for `NextFuture`?

Reply:
We can for sure eliminate `'a` here, as `type NextFuture<'a> = impl Future<Output = StorageResult<Option<DataChunk>>>;` will automatically `+ 'a` when being put in the return position. Anyway, a `+ 'a` here can make reader less confusing about: "why there is an `'a` but not used?"